Rails 5.1 final notes

dhh opened the pull request.

Commit:
Rails 5.1 final notes

prathamesh-sonpatki commented on _posts/2017-04-27-Rails-5-1-final.markdown:
```diff
@@ -0,0 +1,108 @@
+published: true
+date: 2017-04-27 09:00:00 -09:00
+---
+In celebration of the 12th [RailsConf](http://railsconf.com) in Phoenix, Austin this week, we're proud to announce that Rails 5.1 is ready in its final form! We've spent [over 4,100 commits since Rails 5.0](https://github.com/rails/rails/compare/v5.0.0...5-1-stable/) making everything EASIER, SIMPLER, and, uhhh, FUNNER? (That's a RailsConf joke).
```
Phoenix, Arizona :)

dhh replied:
Haha, YES!

dhh replied:
Should link release notes too. Got link?

prathamesh-sonpatki replied:
http://guides.rubyonrails.org/5_1_release_notes.html The link is redirecting to edgeguides now, but will work after the release.

Commit:
Arizona, dammit!

Commit:
Add link to release notes

robin850 commented on _posts/2017-04-27-Rails-5-1-final.markdown:
```diff
@@ -0,0 +1,110 @@
+
+2. **Optionally compile JavaScript with Webpack.** While there are a million different module bundlers/compilers for JavaScript, [Webpack](https://webpack.github.io/) is quickly emerging as the preeminent choice. We've made it easy to use Webpack with Rails through the new [Webpacker](https://github.com/rails/webpacker) gem that you can configure automatically on new projects with `--webpack` (or ...
+
+3. **Drop jQuery as a default dependency.** We used to require jQuery in order to provide features like data-remote, data-confirm, and the other parts of Rails UJS. This dependency is no longer necessary as we've [rewritten rails-ujs to use vanilla JavaScript](https://github.com/rails/rails-ujs). You're of course still free to use jQuery, but you no longer have to.
```
You should rather link to https://github.com/rails/rails/tree/master/actionview/app/assets/javascripts I guess.

kaspth replied:
Think we'll eventually need a better location than GitHub for our JS docs. — @fxn should we just turn our JS docs into guides?

I'll update the link, since it's more correct than the rails/rails-ujs repo after the inlining.

fxn replied:
@kaspth what are the JS docs?

kaspth replied:
@fxn the bits here: https://github.com/rails/rails/tree/master/actionview/app/assets/javascripts

Though I guess an even better link would be the improved working with JavaScript guide: https://github.com/rails/rails/pull/28904 (which will be up at http://guides.rubyonrails.org/working_with_javascript_in_rails.html after the release)

fxn replied:
@kaspth agreed, could go there. Since this is a bit technical, I guess it could go towards the bottom of the guide.

kaspth committed:
Use a GitHub link for rails-ujs JS docs.

Better than the now defunct rails/rails-ujs repo.

kaspth committed:
Anchor the rails-ujs docs to the guide.

Look to http://edgeguides.rubyonrails.org/working_with_javascript_in_rails.html#unobtrusive-javascript
to see what it'll be after the release.

fxn commented:
I have seen that the webpack project consistently writes "webpack" in lowercase. It does even in headers and start of paragraph.

rafaelfranca committed:
Use the final links

rafaelfranca committed:
webpack not Webpack

rafaelfranca commented:
@fxn fixed
